src/main/java/com/powsybl/openloadflow/network/LfContingency.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.powsybl.openloadflow.graph.GraphDecrementalConnectivity;
+import com.powsybl.openloadflow.util.PropagatedContingency;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class LfContingency {
+
+    private final String id;
+
+    private final int index;
+
+    private final Set<LfBus> buses;
+
+    private final Set<LfBranch> branches;
+
+    private double activePowerLoss;
+
+    public LfContingency(String id, int index, Set<LfBus> buses, Set<LfBranch> branches) {
+        this.id = Objects.requireNonNull(id);
+        this.index = index;
+        this.buses = Objects.requireNonNull(buses);
+        this.branches = Objects.requireNonNull(branches);
+        double lose = 0;
+        for (LfBus bus : buses) {
+            lose += bus.getGenerationTargetP() - bus.getLoadTargetP();
+        }
+        this.activePowerLoss = lose;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public int getIndex() {
+        return index;
+    }
+
+    public Set<LfBus> getBuses() {
+        return buses;
+    }
+
+    public Set<LfBranch> getBranches() {
+        return branches;
+    }
+
+    public double getActivePowerLoss() {
+        return activePowerLoss;
+    }
+
+    public static Optional<LfContingency> create(PropagatedContingency propagatedContingency, LfNetwork network,
+                                                 GraphDecrementalConnectivity<LfBus> connectivity, boolean useSmallComponents) {
+        // find contingency branches that are part of this network
+        Set<LfBranch> branches = new HashSet<>(1);
+        for (String branchId : propagatedContingency.getBranchIdsToOpen()) {
```

Comment:
Oh, good not to remove the processed branches from the list nor the PropagatedContingency! didn't really like it before